### PR TITLE
Android-Lint fixes

### PR DESCRIPTION
**Pull request by SleeplessByte**
This includes #32, so you can ignore that if you are going to merge this, or merge that first and then this.

- [Assertion are not supported on Dalvik](http://tools.android.com/recent/androidstudio045released)
- [Try not to use iterators](http://developer.android.com/training/articles/perf-tips.html) or `for( int i )` loops
- Mark exceptions as purposefully unhandled

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/ServerRequestQueue.java [hunk 1]
```diff
@@ -44,14 +48,13 @@ private void persist() {
...
-				synchronized(queue) {
-					Iterator<ServerRequest> iter = queue.iterator();
-					while (iter.hasNext()) {
-						JSONObject json = iter.next().toJSON();
-						if (json != null) {
-							jsonArr.put(json);
-						}
-					}
+				synchronized(lock) {
+                    for ( ServerRequest aQueue : queue ) {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/ServerRequestQueue.java]: same here!

Comment (SleeplessByte) [inline Branch-SDK/src/io/branch/referral/ServerRequestQueue.java]: Ref. https://github.com/BranchMetrics/Branch-Android-SDK/pull/34#discussion_r23390512

**Comment by SleeplessByte** [comment]
As #32 now has moar, this branch can no longer be auto-merged.

File: Branch-SDK/src/io/branch/referral/ServerRequestQueue.java (hunk 1 above)
Comment (SleeplessByte) [inline Branch-SDK/src/io/branch/referral/ServerRequestQueue.java]: d116945

**Comment by qinweigong** [comment]
merged to master

**Pull request closed without merging**